src/effects/effectslot.cpp

```diff
@@ -1,252 +1,528 @@
+}
 
-        while (static_cast<unsigned int>(m_buttonParameters.size())
-                < pEffect->numButtonParameters()) {
-            addEffectButtonParameterSlot();
+void EffectSlot::loadEffectInner(const EffectManifestPointer pManifest,
+        EffectPresetPointer pEffectPreset,
+        bool adoptMetaknobFromPreset) {
+    if (kEffectDebugOutput) {
+        if (pManifest) {
+            qDebug() << this << m_group << "loading effect" << pManifest->id();
+        } else {
+            qDebug() << this << m_group << "unloading effect";
         }
+    }
+    unloadEffect();
+
+    m_pManifest = pManifest;
+
+    if (!pManifest || !pEffectPreset) {
+        // No new effect to load; just unload the old effect and return.
+        emit effectChanged();
+        return;
+    }
 
-        for (const auto& pParameter : m_parameters) {
-            pParameter->loadEffect(pEffect);
+    addToEngine();
+
+    // Create EffectParameters. Every parameter listed in the manifest must have
+    // an EffectParameter created, regardless of whether it is loaded in a slot.
+    for (const auto& pManifestParameter : m_pManifest->parameters()) {
+        // match the manifest parameter to the preset parameter
+        EffectParameterPreset parameterPreset = EffectParameterPreset();
+        if (pEffectPreset) {
+            for (const auto& p : pEffectPreset->getParameterPresets()) {
+                if (p.id() == pManifestParameter->id()) {
+                    parameterPreset = p;
+                }
+            }
         }
-        for (const auto& pParameter : m_buttonParameters) {
-            pParameter->loadEffect(pEffect);
+        EffectParameterPointer pParameter(new EffectParameter(m_pEngineEffect,
+                m_pMessenger,
+                pManifestParameter,
+                parameterPreset));
+        m_allParameters[pManifestParameter->parameterType()].append(pParameter);
+    }
+
+    // Map the parameter slots to the EffectParameters.
+    // The slot order is determined by the order parameters are listed in the preset.
+    int numTypes = static_cast<int>(EffectParameterType::NUM_TYPES);
+    for (int parameterTypeId = 0; parameterTypeId < numTypes;
+            ++parameterTypeId) {
+        const EffectParameterType parameterType =
+                static_cast<EffectParameterType>(parameterTypeId);
+
+        if (pEffectPreset && !pEffectPreset.isNull()) {
+            m_loadedParameters[parameterType].clear();
+            int slot = 0;
+            for (const auto& parameterPreset :
+                    pEffectPreset->getParameterPresets()) {
+                if (parameterPreset.hidden() || parameterPreset.isNull()) {
+                    continue;
+                }
+
+                for (const auto& pParameter :
+                        m_allParameters.value(parameterType)) {
+                    if (pParameter->manifest()->id() == parameterPreset.id()) {
+                        m_loadedParameters[parameterType].insert(
```

Comment:
Is it possible that it tries to insert here with slot > 0 even though the list is empty?

Reply:
I changed this to use QList::append in 7333215. Does that solve the warning for you?

Reply:
Yes, it solves the warning when scrolling through the combobox in the Tango Skin. Thanks.